halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/ClouddriverProfileFactory.java

```diff
@@ -85,6 +80,14 @@ protected void setProfile(
     if (deploymentConfiguration.getProviders() != null) {
       processProviders(deploymentConfiguration.getProviders());
     }
+    Features features = deploymentConfiguration.getFeatures();
```

Comment:
It looks like this PR is creating two places in the hal config to set whether cloudformation is enabled:
* `features.cloudFormation`
* `providers.aws.features.cloudFormation`

The second of these (`providers.aws.features.cloudFormation`) is where we actually want the setting to live as it's where it ends up in the clouddriver config output; this code here looks to be looking at the first location and updating the second to match it.

I don't think we need the `features.cloudFormation` at all; I think that the single source of truth for the value of the setting should be `providers.aws.features.cloudFormation`, which removes the need for this code here and avoids confusingly having two places in the hal config to set it.

In addition to removing the field `features.cloudFormation`, I think you'll also need to change where you've added the command; this should probably be something like `hal config provider aws features ...` and should be registered in `AwsCommand.java`.

Reply:
@ezimanyi Thanks for the feedback!
I agree with you on the changes to be made.
As soon as I have time I will implement them. Thanks again!

Reply:
@ezimanyi could you take a look to the new solution?  Thank you!